src/main/java/io/lettuce/core/RedisClient.java

```diff
@@ -322,6 +323,7 @@ public static RedisClient create(ClientResources clientResources, RedisURI redis
         connectionBuilder(getSocketAddressSupplier(redisURI), connectionBuilder, redisURI);
         connectionBuilder.connectionInitializer(createHandshake(state));
         channelType(connectionBuilder, redisURI);
+        resolver(connectionBuilder, redisURI);
```

Comment:
[Supplemental Comments]
I added `resolver(...)` all places where `channelType(...)` is used.

Reply:
I think it makes sense to refactor the connect code a bit to pull things more together. I'll take care of this during the merge.